src/org/zaproxy/zap/extension/alert/ExtensionAlert.java

```diff
@@ -706,7 +706,7 @@ public String getXml(SiteNode site) {
     }
     
     private String alertFingerprint(Alert alert) {
-    	return alert.getPluginId() + "/" + alert.getRisk() + "/" + alert.getConfidence();
+    	return alert.getPluginId() + "/" + alert.getName() + "/" + alert.getRisk() + "/" + alert.getConfidence();

```

Comment:
Consider adding this in another commit, it's easier to note the (important) change/fix.